dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -92,6 +96,8 @@
             if (maybeResult.isPresent()) {
                 ListenableFuture<Response> result = maybeResult.get();
                 DialogueFutures.addDirectListener(result, this::onCompletion);
+                // While the queue was avoid, this is equivalent to spending zero time on the queue.
+                queuedTime.update(0, TimeUnit.NANOSECONDS);
```

Comment:
I could go either way on this, but I opted toward updating with zero time to avoid biasing our statistics toward problematic regions. This code path is an optimization to avoid unnecessary work in the queue unless requests are already queued.